src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/TerminalSessionSocket.java

```diff
@@ -162,24 +175,114 @@ private void initialize(UIPrefs uiPrefs)
     * an rsession has already been started via RPC and the consoleProcess
     * received.
     * @param consoleProcess 
-    * @param callback returns true if using websockets, false for RPC
+    * @param callback result of connect attempt
     */
    public void connect(ConsoleProcess consoleProcess, 
-                       ServerRequestCallback<Boolean> callback)
+                       final ConnectCallback callback)
    {
       consoleProcess_ = consoleProcess;
-      
+
       // We keep this handler connected after a disconnect so
       // user input sent via RPC can wake up a suspended session
       if (terminalInputHandler_ == null)
          terminalInputHandler_ = xterm_.addTerminalDataInputHandler(this);
 
       addHandlerRegistration(consoleProcess_.addConsoleOutputHandler(this));
-      
-      // TODO (gary) attempt websocket connection here
-      boolean haveWebsocket = false;
-      
-      callback.onResponseReceived(haveWebsocket);
+
+      switch (consoleProcess_.getChannelMode())
+      {
+      case ConsoleProcessInfo.CHANNEL_RPC:
+         Debug.devlog("Using RPC");
```

Comment:
Usually we only use `devlog` for debugging statements that aren't intended to be committed -- can you remove this or change it to another debug logging method?